Sources/OpenAPIServiceModel/CreateOpenAPIServiceModel.swift

```diff
@@ -0,0 +1,382 @@
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+//
+// CreateOpenAPIServiceModel.swift
+// OpenAPIServiceModel
+//
+
+import Foundation
+import OpenAPIKit
+import ServiceModelEntities
+import ServiceModelCodeGeneration
+import Yams
+internal extension OpenAPIServiceModel {
+    struct OperationInputMembers {
+        var queryMembers: [String: Member] = [:]
+        var additionalHeaderMembers: [String: Member] = [:]
+        var pathMembers: [String: Member] = [:]
+    }
+    
+    static func filterOperations(operations: [OpenAPI.HttpMethod: OpenAPI.Operation],
+                                 modelOverride: ModelOverride?) -> [OpenAPI.HttpMethod: OpenAPI.Operation] {
+        
+        guard let ignoreOperations = modelOverride?.ignoreOperations else {
+            return operations
+        }
+        
+        var filteredOperations: [OpenAPI.HttpMethod: OpenAPI.Operation] = [:]
+        
+        operations.forEach { (key, value) in
+            if ignoreOperations.contains("*.*") {
+                return
+            }
+            
+            if ignoreOperations.contains("*.\(key.rawValue)") {
+                return
+            }
+            
+            if let identifier = value.operationId {
+                if ignoreOperations.contains("\(identifier).\(key.rawValue)") {
+                    return
+                }
+                
+                if ignoreOperations.contains("\(identifier).*") {
+                    return
+                }
+            }
+            
+            filteredOperations[key] = value
+        }
+        
+        return filteredOperations
+    }
+    
+    static func createOpenAPIModel(definition: OpenAPI.Document, modelOverride: ModelOverride?) -> OpenAPIServiceModel {
+        var model = OpenAPIServiceModel()
+
+        for (name, schema) in definition.components.schemas {
+            var enclosingEntityName = name.rawValue
+            parseDefinitionSchemas(model: &model, enclosingEntityName: &enclosingEntityName,
+                                   schema: schema, modelOverride: modelOverride, document: definition)
+        }
+        
+        for (path, pathDefinition) in definition.paths {
+            var operations:[OpenAPI.HttpMethod: OpenAPI.Operation] = [:]
+            
+            pathDefinition.endpoints.forEach { endpoint in
+                operations[endpoint.method] = endpoint.operation
+            }
+            
+            let filteredOperations = filterOperations(operations: operations,
+                                                      modelOverride: modelOverride)
+            
+            for (type, operation) in filteredOperations {
+                guard let operationName = operation.operationId else {
+                    continue
```

Comment:
should we ever hit this else block?  If not maybe we can throw here instead?

Reply:
operationID is optional so I think it may best to leave the continue.